server/src/test/java/org/elasticsearch/index/codec/CodecTests.java

```diff
@@ -148,4 +151,11 @@ private CodecService createCodecService() throws IOException {
         return new CodecService(service, BigArrays.NON_RECYCLING_INSTANCE);
     }
 
+    private static Codec unwrappedCodec(CodecService codecService, String codecName) {
+        Codec codec = codecService.codec(codecName);
+        if (codec instanceof DeduplicateFieldInfosCodec deduplicatingCodec) {
```

Comment:
I think this can be:
```suggestion
        if (codec instanceof FilterCoded filtered) {
```

Reply:
Unfortunately not, the `delegate` is hidden from us there.